test/adapters/managing.test.js

```diff
@@ -652,12 +712,241 @@ describe("Adapter - Managing", () => {
         flags: 0,
+        keys: [],
+        values: [],
+        extensionAddresses: [],
+        extensionAclFlags: [],
       },
       [],
+      {
+        from: daoOwner,
+        gasPrice: toBN("0"),
+      }
+    );
+
+    await voting.submitVote(dao.address, proposalId, 1, {
+      from: daoOwner,
+      gasPrice: toBN("0"),
+    });
+
+    await advanceTime(1000);
+
+    await expectRevert(
+      managing.processProposal(dao.address, proposalId, {
+        from: daoOwner,
+        gasPrice: toBN("0"),
+      }),
+      "adapterAddress already in use"
+    );
+  });
+
+  it("should be possible to add a new adapter and set the acl flags for some extension", async () => {
+    const dao = this.dao;
+    const managing = this.adapters.managing;
+    const voting = this.adapters.voting;
+    const financing = await FinancingContract.new();
+    const bankExt = this.extensions.bank;
+
+    const newAdapterId = sha3("testFinancing");
+    const newAdapterAddress = financing.address;
+    const proposalId = getProposalCounter();
+
+    await managing.submitProposal(
+      dao.address,
+      proposalId,
+      {
+        adapterOrExtensionId: newAdapterId,
+        adapterOrExtensionAddr: newAdapterAddress,
+        updateType: 1,
+        flags: 0,
+        keys: [],
+        values: [],
+        // Set the extension address which will be accessed by the new adapter
+        extensionAddresses: [bankExt.address],
+        // Set the acl flags so the new adapter can access the bank extension
+        extensionAclFlags: [
+          entryBank(financing, {
+            ADD_TO_BALANCE: true,
+            SUB_FROM_BALANCE: true,
+            INTERNAL_TRANSFER: true,
+          }).flags,
+        ],
+      },
       [],
+      {
+        from: daoOwner,
+        gasPrice: toBN("0"),
+      }
+    );
+
+    await voting.submitVote(dao.address, proposalId, 1, {
+      from: daoOwner,
+      gasPrice: toBN("0"),
+    });
+
+    await advanceTime(1000);
+
+    await managing.processProposal(dao.address, proposalId, {
+      from: daoOwner,
+      gasPrice: toBN("0"),
+    });
+
+    // At this point the adapter should be able access the Bank Extension
+    // We check that by verifying if the ACL flag in the DAO matches the one
+    // submitted in the proposal.
+    expect(await dao.getAdapterAddress(newAdapterId)).equal(newAdapterAddress);
+    expect(
+      await dao.hasAdapterAccessToExtension(
+        newAdapterAddress,
+        bankExt.address,
+        entryBank(financing, { ADD_TO_BALANCE: true }).flags
+      )
+    ).equal(true);
+    expect(
+      await dao.hasAdapterAccessToExtension(
+        newAdapterAddress,
+        bankExt.address,
+        entryBank(financing, { SUB_FROM_BALANCE: true }).flags
+      )
+    ).equal(true);
+    expect(
+      await dao.hasAdapterAccessToExtension(
+        newAdapterAddress,
+        bankExt.address,
+        entryBank(financing, { INTERNAL_TRANSFER: true }).flags
+      )
+    ).equal(true);
```

Comment:
For some reason this condition is not passing. I will continue to investigate that.

Reply:
I will check too this morning

Reply:
fixed!